pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -36,8 +38,66 @@ def __init__(self, runtime: Runtime, version: str, assembly: str, publish: bool,
         self.logger = runtime.logger
         self.working_dir = self.runtime.working_dir
         self.fail_count_name = f'count:build-sync-failure:{assembly}:{version}'
+        self.job_run = get_build_url()
+
+    async def comment_on_assembly_pr(self, text_body):
+        """
+        Comment the link to this jenkins build on the assembly PR if it was triggered automatically
+        """
+        # Keeping in try-except so that job doesn't fail because of any error here
+        try:
+            base_repo = "openshift-eng"
+            repository = "ocp-build-data"
+            branch = self.doozer_data_gitref
+            token = os.environ.get('GITHUB_TOKEN')
+
+            pattern = r"github\.com/([^/]+)/"
+            match = re.search(pattern, self.data_path)
+
+            repo_owner = None
+            if match:
+                repo_owner = match.group(1)
+
+            api = GhApi(owner=base_repo, repo=repository, token=token)
+
+            # Check if the doozer_data_gitref is given then, if not
+            # then it set the branch to openshift-{major}.{minor}
+            if not branch:
+                branch = f"openshift-{self.version}"
+
+            # Head needs to have the repo name prepended for GhApi to fetch the correct one
+            head = f"{repo_owner}:{branch}"
+            # Find our assembly PR.
+            prs = api.pulls.list(head=head, state="open")
+
+            if len(prs) != 1:
+                self.logger.error(
+                    f"{len(prs)} PR(s) were found with head={head}. We need only 1.")
+                return
+
+            if len(prs) == 0:
+                self.logger.error(f"No assembly PRs were found with head={head}")
+                return
+
+            pr_number = prs[0]["number"]
```

Comment:
the comment on pr should happen in gen_assembly and will be more easily get the context we can get the pr_number after pr created

Reply:
I think this would mean making build-sync blocking when triggered from gen-assembly. This is probably undesirable